lib/imapSimple.js

```diff
@@ -225,9 +261,49 @@ function connect(options, callback) {
     });
 }
 
+/**
+ * Returns a list of the parts.
+ * A part is either a message or an attachment.
+ * Parts are usually nested and numbered things like 1, 1.1, 1.2, ...
+ * Code taken from http://stackoverflow.com/questions/25247207/how-to-read-and-save-attachments-using-node-imap
+ */
+function findParts(struct, parts) {
```

Comment:
Sure. `attributes.struct` will only exist if the user uses `{ struct: true }` in their fetch options, but if it's in the message, then we can call `getParts(attributes.struct)` in the messageOnEnd() handler and save the result to the corresponding part on a field `struct`. The result would then look like:

```js
{
  attributes: {
    // ...
    struct: { /* ... */ },
  },
  parts: [
    { which: '', size: 123, body: /* decodedBody */, struct: { /* ... */ }, 
    { /* ... */ } 
  ]
}
```